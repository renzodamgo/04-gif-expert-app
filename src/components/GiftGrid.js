// import React, { useEffect, useState } from 'react'
// import { getGifts } from '../helpers/getGifs'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types';

export const GiftGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs(category);
    console.log(images,loading)

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'> {category}</h3>
            {loading && <p>Loading....</p>}

            <div className='card-grid'>

                {
                    images.map((img) =>
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    )
                }



            </div>

        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}
