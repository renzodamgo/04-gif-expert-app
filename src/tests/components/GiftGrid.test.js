import { GiftGrid } from "../../components/GiftGrid";
import { shallow } from 'enzyme';
import { useFetchGifs } from "../../hooks/useFetchGifs";
import '@testing-library/jest-dom';
jest.mock("../../hooks/useFetchGifs");

describe('pruebas en el GiftGrid', () => {

    const category = 'Digimon'
    test('debe mostraste correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GiftGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    })
    test('debe mostrar items cuendo se cargan imÃ¡genes', () => {

        const imgs = [{
            id: 'ABC',
            url: 'https://localhost:3000/cualquier/cosa',
            title:'Caulquiera'
        },
            {
                id: 'ESS',
                url: 'https://localhost:3000/cualquier/cosa',
                title: 'Otra imagen'
            }]
        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        })
        const wrapper = shallow(<GiftGrid category={category} />)
        expect(wrapper.find('p').exists()).toBe(false)
        // expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length)
    })
})