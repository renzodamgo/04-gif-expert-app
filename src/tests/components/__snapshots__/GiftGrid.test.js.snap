// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el GiftGrid debe mostraste correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Digimon
  </h3>
  <p>
    Loading....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
